runelite-plugin-archetype tests are broken

Currently, when there is any change in runelite on what ExamplePlugin (or other classes in the runelite-plugin-archetype project) depends on, the tests start failing, so f.e. PR containing such changes would never pass Travis CI verification.

Here is example output from #351: https://api.travis-ci.org/v3/job/328218822/log.txt
The most important part from that log:
```
[[1;34mINFO[m] Processing Archetype IT project: compilationtest
[[1;34mINFO[m] ----------------------------------------------------------------------------
[[1;34mINFO[m] Using following parameters for creating project from Archetype: runelite-plugin-archetype:1.2.12-SNAPSHOT
[[1;34mINFO[m] ----------------------------------------------------------------------------
[[1;34mINFO[m] Parameter: groupId, Value: org.example.runelite
[[1;34mINFO[m] Parameter: artifactId, Value: exampleplugin
[[1;34mINFO[m] Parameter: version, Value: 1.0.0-SNAPSHOT
[[1;34mINFO[m] Parameter: package, Value: org.example.runelite.exampleplugin
[[1;34mINFO[m] Parameter: packageInPathFormat, Value: org/example/runelite/exampleplugin
[[1;34mINFO[m] Parameter: package, Value: org.example.runelite.exampleplugin
[[1;34mINFO[m] Parameter: version, Value: 1.0.0-SNAPSHOT
[[1;34mINFO[m] Parameter: sourceEncoding, Value: UTF-8
[[1;34mINFO[m] Parameter: groupId, Value: org.example.runelite
[[1;34mINFO[m] Parameter: artifactId, Value: exampleplugin
[[1;34mINFO[m] Project created from Archetype in dir: /home/travis/build/runelite/runelite/runelite-plugin-archetype/target/test-classes/projects/compilationtest/project/exampleplugin
[[1;34mINFO[m] Invoking post-archetype-generation goals: clean verify
[[1;34mINFO[m] Picked up _JAVA_OPTIONS: -Xmx2048m -Xms512m
[[1;34mINFO[m] [INFO] Error stacktraces are turned on.
[[1;34mINFO[m] [INFO] Scanning for projects...
[[1;34mINFO[m] [INFO] 
[[1;34mINFO[m] [INFO] ------------------------------------------------------------------------
[[1;34mINFO[m] [INFO] Building exampleplugin 1.0.0-SNAPSHOT
[[1;34mINFO[m] [INFO] ------------------------------------------------------------------------
[[1;34mINFO[m] [INFO] Downloading from apache-snapshots: https://repository.apache.org/snapshots/net/runelite/client/1.2.12-SNAPSHOT/maven-metadata.xml
[[1;34mINFO[m] [INFO] Downloading from runelite: http://repo.runelite.net/net/runelite/client/1.2.12-SNAPSHOT/maven-metadata.xml
[[1;34mINFO[m] [INFO] Downloading from sonatype-snapshots: https://oss.sonatype.org/content/repositories/snapshots/net/runelite/client/1.2.12-SNAPSHOT/maven-metadata.xml
[[1;34mINFO[m] [INFO] Downloaded from runelite: http://repo.runelite.net/net/runelite/client/1.2.12-SNAPSHOT/maven-metadata.xml (1.2 kB at 3.3 kB/s)
[[1;34mINFO[m] [INFO] Downloading from sonatype-snapshots: https://oss.sonatype.org/content/repositories/snapshots/net/runelite/client/1.2.12-SNAPSHOT/client-1.2.12-20180112.021014-8.pom
[[1;34mINFO[m] [INFO] Downloading from apache-snapshots: https://repository.apache.org/snapshots/net/runelite/client/1.2.12-SNAPSHOT/client-1.2.12-20180112.021014-8.pom
[[1;34mINFO[m] [INFO] Downloading from runelite: http://repo.runelite.net/net/runelite/client/1.2.12-SNAPSHOT/client-1.2.12-20180112.021014-8.pom
[[1;34mINFO[m] [INFO] Downloaded from runelite: http://repo.runelite.net/net/runelite/client/1.2.12-SNAPSHOT/client-1.2.12-20180112.021014-8.pom (7.2 kB at 74 kB/s)
[[1;34mINFO[m] [INFO] Downloading from sonatype-snapshots: https://oss.sonatype.org/content/repositories/snapshots/net/runelite/runelite-parent/1.2.12-SNAPSHOT/maven-metadata.xml
[[1;34mINFO[m] [INFO] Downloading from apache-snapshots: https://repository.apache.org/snapshots/net/runelite/runelite-parent/1.2.12-SNAPSHOT/maven-metadata.xml
[[1;34mINFO[m] [INFO] Downloading from runelite: http://repo.runelite.net/net/runelite/runelite-parent/1.2.12-SNAPSHOT/maven-metadata.xml
[[1;34mINFO[m] [INFO] Downloaded from runelite: http://repo.runelite.net/net/runelite/runelite-parent/1.2.12-SNAPSHOT/maven-metadata.xml (606 B at 6.8 kB/s)
[[1;34mINFO[m] [INFO] Downloading from runelite: http://repo.runelite.net/net/runelite/runelite-parent/1.2.12-SNAPSHOT/runelite-parent-1.2.12-20180112.020458-8.pom
[[1;34mINFO[m] [INFO] Downloaded from runelite: http://repo.runelite.net/net/runelite/runelite-parent/1.2.12-SNAPSHOT/runelite-parent-1.2.12-20180112.020458-8.pom (0 B at 0 B/s)
[[1;34mINFO[m] [INFO] Downloading from apache-snapshots: https://repository.apache.org/snapshots/net/runelite/api/1.2.12-SNAPSHOT/maven-metadata.xml
[[1;34mINFO[m] [INFO] Downloading from apache-snapshots: https://repository.apache.org/snapshots/net/runelite/rs/api/1.2.12-SNAPSHOT/maven-metadata.xml
[[1;34mINFO[m] [INFO] Downloading from sonatype-snapshots: https://oss.sonatype.org/content/repositories/snapshots/net/runelite/injected-client/1.2.12-SNAPSHOT/maven-metadata.xml
[[1;34mINFO[m] [INFO] Downloading from apache-snapshots: https://repository.apache.org/snapshots/net/runelite/injected-client/1.2.12-SNAPSHOT/maven-metadata.xml
[[1;34mINFO[m] [INFO] Downloading from runelite: http://repo.runelite.net/net/runelite/injected-client/1.2.12-SNAPSHOT/maven-metadata.xml
[[1;34mINFO[m] [INFO] Downloaded from runelite: http://repo.runelite.net/net/runelite/injected-client/1.2.12-SNAPSHOT/maven-metadata.xml (780 B at 8.6 kB/s)
[[1;34mINFO[m] [INFO] Downloading from sonatype-snapshots: https://oss.sonatype.org/content/repositories/snapshots/net/runelite/injected-client/1.2.12-SNAPSHOT/injected-client-1.2.12-20180112.020947-8.pom
[[1;34mINFO[m] [INFO] Downloading from apache-snapshots: https://repository.apache.org/snapshots/net/runelite/injected-client/1.2.12-SNAPSHOT/injected-client-1.2.12-20180112.020947-8.pom
[[1;34mINFO[m] [INFO] Downloading from runelite: http://repo.runelite.net/net/runelite/injected-client/1.2.12-SNAPSHOT/injected-client-1.2.12-20180112.020947-8.pom
[[1;34mINFO[m] [INFO] Downloaded from runelite: http://repo.runelite.net/net/runelite/injected-client/1.2.12-SNAPSHOT/injected-client-1.2.12-20180112.020947-8.pom (3.0 kB at 38 kB/s)
[[1;34mINFO[m] [INFO] Downloading from sonatype-snapshots: https://oss.sonatype.org/content/repositories/snapshots/net/runelite/http-api/1.2.12-SNAPSHOT/maven-metadata.xml
[[1;34mINFO[m] [INFO] Downloading from apache-snapshots: https://repository.apache.org/snapshots/net/runelite/http-api/1.2.12-SNAPSHOT/maven-metadata.xml
[[1;34mINFO[m] [INFO] Downloading from runelite: http://repo.runelite.net/net/runelite/http-api/1.2.12-SNAPSHOT/maven-metadata.xml
[[1;34mINFO[m] [INFO] Downloaded from runelite: http://repo.runelite.net/net/runelite/http-api/1.2.12-SNAPSHOT/maven-metadata.xml (988 B at 11 kB/s)
[[1;34mINFO[m] [INFO] Downloading from sonatype-snapshots: https://oss.sonatype.org/content/repositories/snapshots/net/runelite/http-api/1.2.12-SNAPSHOT/http-api-1.2.12-20180112.020505-7.pom
[[1;34mINFO[m] [INFO] Downloading from apache-snapshots: https://repository.apache.org/snapshots/net/runelite/http-api/1.2.12-SNAPSHOT/http-api-1.2.12-20180112.020505-7.pom
[[1;34mINFO[m] [INFO] Downloading from runelite: http://repo.runelite.net/net/runelite/http-api/1.2.12-SNAPSHOT/http-api-1.2.12-20180112.020505-7.pom
[[1;34mINFO[m] [INFO] Downloaded from runelite: http://repo.runelite.net/net/runelite/http-api/1.2.12-SNAPSHOT/http-api-1.2.12-20180112.020505-7.pom (2.7 kB at 34 kB/s)
[[1;34mINFO[m] [INFO] Downloading from runelite: http://repo.runelite.net/net/runelite/client/1.2.12-SNAPSHOT/client-1.2.12-20180112.021014-8.jar
[[1;34mINFO[m] [INFO] Downloading from runelite: http://repo.runelite.net/net/runelite/injected-client/1.2.12-SNAPSHOT/injected-client-1.2.12-20180112.020947-8.jar
[[1;34mINFO[m] [INFO] Downloading from runelite: http://repo.runelite.net/net/runelite/http-api/1.2.12-SNAPSHOT/http-api-1.2.12-20180112.020505-7.jar
[[1;34mINFO[m] [INFO] Downloaded from runelite: http://repo.runelite.net/net/runelite/client/1.2.12-SNAPSHOT/client-1.2.12-20180112.021014-8.jar (753 kB at 2.6 MB/s)
[[1;34mINFO[m] [INFO] Downloaded from runelite: http://repo.runelite.net/net/runelite/injected-client/1.2.12-SNAPSHOT/injected-client-1.2.12-20180112.020947-8.jar (1.6 MB at 4.7 MB/s)
[[1;34mINFO[m] [INFO] Downloaded from runelite: http://repo.runelite.net/net/runelite/http-api/1.2.12-SNAPSHOT/http-api-1.2.12-20180112.020505-7.jar (52 kB at 62 kB/s)
[[1;34mINFO[m] [INFO] 
[[1;34mINFO[m] [INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ exampleplugin ---
[[1;34mINFO[m] [INFO] 
[[1;34mINFO[m] [INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ exampleplugin ---
[[1;34mINFO[m] [INFO] Using 'UTF-8' encoding to copy filtered resources.
[[1;34mINFO[m] [INFO] skip non existing resourceDirectory /home/travis/build/runelite/runelite/runelite-plugin-archetype/target/test-classes/projects/compilationtest/project/exampleplugin/src/main/resources
[[1;34mINFO[m] [INFO] 
[[1;34mINFO[m] [INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ exampleplugin ---
[[1;34mINFO[m] [INFO] Changes detected - recompiling the module!
[[1;34mINFO[m] [INFO] Compiling 3 source files to /home/travis/build/runelite/runelite/runelite-plugin-archetype/target/test-classes/projects/compilationtest/project/exampleplugin/target/classes
[[1;34mINFO[m] [INFO] -------------------------------------------------------------
[[1;34mINFO[m] [ERROR] COMPILATION ERROR : 
[[1;34mINFO[m] [INFO] -------------------------------------------------------------
[[1;34mINFO[m] [ERROR] /home/travis/build/runelite/runelite/runelite-plugin-archetype/target/test-classes/projects/compilationtest/project/exampleplugin/src/main/java/org/example/runelite/exampleplugin/ExamplePlugin.java:[14,35] cannot find symbol
[[1;34mINFO[m]   symbol:   class PluginConfig
[[1;34mINFO[m]   location: package net.runelite.client.plugins
[[1;34mINFO[m] [ERROR] /home/travis/build/runelite/runelite/runelite-plugin-archetype/target/test-classes/projects/compilationtest/project/exampleplugin/src/main/java/org/example/runelite/exampleplugin/ExamplePluginConfiguration.java:[4,35] cannot find symbol
[[1;34mINFO[m]   symbol:   class PluginConfig
[[1;34mINFO[m]   location: package net.runelite.client.plugins
[[1;34mINFO[m] [ERROR] /home/travis/build/runelite/runelite/runelite-plugin-archetype/target/test-classes/projects/compilationtest/project/exampleplugin/src/main/java/org/example/runelite/exampleplugin/ExamplePluginConfiguration.java:[6,53] cannot find symbol
[[1;34mINFO[m]   symbol: class PluginConfig
[[1;34mINFO[m] [ERROR] /home/travis/build/runelite/runelite/runelite-plugin-archetype/target/test-classes/projects/compilationtest/project/exampleplugin/src/main/java/org/example/runelite/exampleplugin/ExamplePlugin.java:[76,27] cannot find symbol
[[1;34mINFO[m]   symbol:   class PluginConfig
[[1;34mINFO[m]   location: class org.example.runelite.exampleplugin.ExamplePlugin
[[1;34mINFO[m] [ERROR] /home/travis/build/runelite/runelite/runelite-plugin-archetype/target/test-classes/projects/compilationtest/project/exampleplugin/src/main/java/org/example/runelite/exampleplugin/ExamplePlugin.java:[25,1] annotation @net.runelite.client.plugins.PluginDescriptor is missing a default value for the element 'name'
[[1;34mINFO[m] [ERROR] /home/travis/build/runelite/runelite/runelite-plugin-archetype/target/test-classes/projects/compilationtest/project/exampleplugin/src/main/java/org/example/runelite/exampleplugin/ExamplePlugin.java:[75,9] method does not override or implement a method from a supertype
[[1;34mINFO[m] [ERROR] /home/travis/build/runelite/runelite/runelite-plugin-archetype/target/test-classes/projects/compilationtest/project/exampleplugin/src/main/java/org/example/runelite/exampleplugin/ExamplePlugin.java:[78,41] method getConfig in class net.runelite.client.config.ConfigManager cannot be applied to given types;
[[1;34mINFO[m]   required: java.lang.Class<T>
[[1;34mINFO[m]   found: java.lang.Class<capture#1 of ? extends org.example.runelite.exampleplugin.ExamplePlugin>,java.lang.Class<org.example.runelite.exampleplugin.ExamplePluginConfiguration>
[[1;34mINFO[m]   reason: cannot infer type-variable(s) T
[[1;34mINFO[m]     (actual and formal argument lists differ in length)
[[1;34mINFO[m] [INFO] 7 errors 
[[1;34mINFO[m] [INFO] -------------------------------------------------------------
[[1;34mINFO[m] [INFO] ------------------------------------------------------------------------
[[1;34mINFO[m] [INFO] BUILD FAILURE
[[1;34mINFO[m] [INFO] ------------------------------------------------------------------------
[[1;34mINFO[m] [INFO] Total time: 4.846 s
[[1;34mINFO[m] [INFO] Finished at: 2018-01-12T18:06:52Z
[[1;34mINFO[m] [INFO] Final Memory: 22M/491M
[[1;34mINFO[m] [INFO] ------------------------------------------------------------------------
[[1;34mINFO[m] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project exampleplugin
```

--- a/runelite-plugin-archetype/src/test/resources/projects/compilationtest/goal.txt
+++ b/runelite-plugin-archetype/src/test/resources/projects/compilationtest/goal.txt
@@ -1,1 +1,1 @@
-clean verify
+verify

--- a/travis/build.sh
+++ b/travis/build.sh
@@ -8,5 +8,5 @@ PROJECT_VERSION=`mvn -q -Dexec.executable="echo" -Dexec.args='${project.version}
 if [[ "${TRAVIS_PULL_REQUEST}" == "false" && $PROJECT_VERSION == *"-SNAPSHOT" && "$TRAVIS_BRANCH" == "master" ]]; then
 	mvn clean deploy --settings travis/settings.xml
 else
-	mvn clean verify --settings travis/settings.xml
+	mvn clean install --settings travis/settings.xml
 fi

